models/classes/LtiProvider/LtiProviderFactory.php

```diff
@@ -71,21 +70,27 @@ public function createFromResource(core_kernel_classes_Resource $resource): LtiP
 
     public function createFromArray(array $provider): LtiProvider
     {
-        $keys = ['uri', 'label', 'key', 'secret', 'callback_url'];
+        $ltiVersion = $provider[ConfigurableLtiProviderRepository::LTI_VERSION] ?: '1.1';
 
-        foreach ($keys as $key) {
-            if (!isset($provider[$key])) {
-                throw new InvalidArgumentException(sprintf('Missing key \'%s\' in LTI provider list.', $key));
-            }
-        }
+        $this->getValidationService()->validateArray($ltiVersion, $provider);
 
         return new LtiProvider(
             $provider['uri'],
             $provider['label'],
             $provider['key'],
             $provider['secret'],
             $provider['callback_url'],
-            $provider['roles'] ?? []
+            $provider['roles'] ?? [],
+            $ltiVersion,
+            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_IDENTIFIER]?:null,
+            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_NAME]?:null,
+            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_CLIENT_ID]?:null,
+            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_DEPLOYMENT_IDS]?:[],
+            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_AUDIENCE]?:null,
+            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_OIDC_LOGIN_INITATION_URL]?:null,
+            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_LAUNCH_URL]?:null,
+            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_PUBLIC_KEY]?:null,
+            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_JWKS_URL]?:null
```

Comment:
```suggestion
            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_IDENTIFIER] ?? null,
            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_NAME] ?? null,
            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_CLIENT_ID] ?? null,
            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_DEPLOYMENT_IDS] ?? [],
            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_AUDIENCE] ?? null,
            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_OIDC_LOGIN_INITATION_URL] ?? null,
            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_LAUNCH_URL] ?? null,
            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_PUBLIC_KEY] ?? null,
            $provider[ConfigurableLtiProviderRepository::LTI_TOOL_JWKS_URL] ?? null
```

Otherwise if the index does not exist it will throw an error. Also good to have spaces for readability :)